pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -433,27 +427,6 @@ func (c *ReverseExpandQuery) execute(
 		}
 	}
 
-	// NOTE: this is temporary to ensure that the ListObjects shadow query doesn't run unnecessarily.
-	// For cases where the query is weight INF, reverse_expand_weighted falls back to original reverse_expand,
-	// so there is no value in running a shadow query after the main query completes.
-	// This block will hit on the first pass through reverse_expand, and it marks ShouldRunShadowQuery based on
-	// whether the shadow query will actually run the code we want to test.
-	if !req.skipWeightedGraph {
-		req.skipWeightedGraph = true // ensure we don't do this on subsequent recursive calls
-		resolutionMetadata.ShouldRunShadowQuery.Store(true)
-
-		typeRel := tuple.ToObjectRelationString(targetObjRef.GetType(), targetObjRef.GetRelation())
-		node, ok := c.typesystem.GetNode(typeRel)
-		if !ok {
-			resolutionMetadata.ShouldRunShadowQuery.Store(false)
-		} else {
-			weight, _ := node.GetWeight(sourceUserType)
-			if weight == weightedGraph.Infinite {
-				resolutionMetadata.ShouldRunShadowQuery.Store(false)
-			}
-		}
-	}
-
```

Comment:
excuse my ignorance 🙂 does this mean the shadow query will always run now, or is it gated some other way?

is that related at all to the PR's benchmark tests failing?